### ISSUE TITLE
Feature request: Always open URLs that match a pattern in a container

### ISSUE THREAD

**Issue by keeferrourke** [mozilla/testpilot-containers#691]
At present, one can mark a site to always be opened in a particular container, and this is incredibly useful for sandboxing certain accounts from one another. For instance, I can always open http://twitter.com in a "Twitter" container, and http://google.com in a "Google" container.

However, I've noticed that opening http://play.google.com and http://music.google.com for instance need to be individually marked to be opened in my "Google" container. It would be really nice if I were able to automatically open all URLs that match a custom pattern such as `*.google.*` in a particular container. This saves adding rules for each subdomain of a website.

**Comment by ransage** [comment]
Similarly, this might enable one to always open "github.com/mycompany" in a work container and "github.com/myHobbyProject" in a HobbyProject container

**Comment by jonathanKingston** [comment]
I think this is actually a duplicate too.

I think this is probably a candidate for PR's perhaps or alternate extensions that behave the same.
Currently assignment is a little buggy as it stands anyway so it is worth waiting for now. I can see this being a great power user feature.

**Comment by dschneiderch** [comment]
> I've noticed that opening http://play.google.com and http://music.google.com for instance need to be individually marked 

This is great feature for me as it already is so I'd vote not to change the current default behavior. I open different google subdomains with dffferent google accounts, i.e. university vs personal.

**Comment by keeferrourke** [comment]
I wasn't suggesting changing the default behaviour, but rather allowing for more advanced pattern matching if one desires. Your use case is as valid as mine :grin:

**Comment by grahamperrin** [comment]
> … duplicate … 

https://github.com/mozilla/multi-account-containers/issues?q=is%3Aissue+sort%3Aupdated-desc+containerise might help to de-duplicate and maybe close an issue. Related: 

- [Containerise](https://addons.mozilla.org/addon/containerise/)

**Comment by knut-hildebrandt** [comment]
Would wish for the support of URLs as well. An use case example is Wordpress. If making changes to a site it would be nice to have the backend in one container while the live site is in another. But after mapping the backend to "work" container the site also always opens in "work" and vice versa. Of course I can temporally choose another container, but making this permanently changes it for the whole (sub)domain and not only the page and its subpages. An good example how this could work better is "Dark Background and Light Text" (https://addons.mozilla.org/de/firefox/addon/dark-background-light-text)

**Comment by tubize** [comment]
This would be a great addition, for instance when working with Azure Devops.
I work for a company that does a lot of projects for other companies, often I need to switch back and forth between their devops and ours. Now I do it in an incognito window, but it would be great if
dev.azure.com/MyCompany     and
dev.azure.com/TheOtherCompany
would open in a different container automatically, keeping me signed in.

**Comment by ChaosFreak** [comment]
Yes, this absolutely would be incredibly useful, especially for opening Google Drive links from different domains in different containers.

**Comment by nikopen** [comment]
folks, the Containerise plugin seems to do what we want: 
https://addons.mozilla.org/en-US/firefox/addon/containerise/

this issue should be closed

/close

**Comment by Mcinwwl** [comment]
Hi, would such enhancement be limited to wildcards, or is there any regex support possible?

**Comment by ChaosFreak** [comment]
> 
> 
> Hi, would such enhancement be limited to wildcards, or is there any regex support possible?

I think regex support would be a nice-to-have, perhaps as an option. In the first version, simple wildcards would probably cover 90% of use cases. Then, after that, we could add full regex support.

**Comment by dscho** [comment]
For anybody interested in this: I worked a bit on this, primarily because I need it for Azure DevOps: https://github.com/dscho/multi-account-containers/commit/a5eb0d6d98f2233cf3dad273bbe8bfcea94af6fb. I have no need for this to be configurable, so that's left to implement (feel free to take this commit and run from there).

**Comment by sashaweiss** [comment]
@dscho I want to do something similar, in that I want to bake in some logic that covers my specific use cases but that might not be general enough to contribute back to the main project.

How did you install/distribute from your fork? Specifically, re: addon signing, and the fact that unsigned addons can't be added to stable Firefox. I want to use stable Firefox, but also want this functionality. However, I'm not sure if a fork with some specifically reduced functionality would qualify for me to submit for signing, since it'd be 99% similar to what exists here.

**Comment by dscho** [comment]
# @sashaweiss I *think* I followed some official documentation and submitted it for signing, stating that it is intended for personal use, not for distributing. I don't recall reading that this is a use case that wouldn't be eligible for signing...

**Comment by sashaweiss** [comment]
Gotcha, thanks for the tip! I’ll try and do the same, and make it clear it’s for personal use.

**Comment by grahamperrin** [comment]
Some overlap with #839?

**Comment by dsfrederic** [comment]
@dscho can you describe how you installed your own modified version? 

I have exactly the same use case.

**Comment by achernyakevich-sc** [comment]
@dsfrederic Mayve this link will help:
https://github.com/mozilla/multi-account-containers/issues/2100#issuecomment-900967202

**Comment by dscho** [comment]
> 
> 
> @dscho can you describe how you installed your own modified version?
> 
> I have exactly the same use case.

I don't recall the details, sorry... I _think_ I had to register as add-on developer (probably somewhere near https://addons.mozilla.org/en-US/developers/, I really forgot the details), get my secrets, sign the package, upload it (I remember that I marked it as private), and then I had to wait a little until it came back signed, or something like that.

**Comment by dwillis77** [comment]
Just curious, this request is now 4+ years old (not to mention several other similar requests that have been closed as duplicates of this - at least #976 , #1254 , #2100 and possibly others) and clearly would be useful to many people; is there a roadblock in implementing this? At one point it seems it was possibly due to lack of active development on this project, but I see now there have been recent releases so that doesn't seem to be an issue anymore. This is an amazing extension that is incredibly useful and adding this feature to it would enhance it greatly!

I know some have said the Containerise plugin serves this purpose (although people seem to have mixed results with that) - is that the reason it is a lower priority to implement the same feature in this project? Or is it because it would be a complex feature to implement? Or something else?

I might follow @dscho 's lead and try to take a stab at this myself... thought it would probably take me quite a while to put something together that was worthy of incorporating into the main project.

**Comment by dscho** [comment]
> I might follow @dscho 's lead and try to take a stab at this myself...

Well, https://github.com/dscho/multi-account-containers/releases/tag/6.2.5-dscho works for me, so it might not take all that much for you to get there.

**Comment by dwillis77** [comment]
Heh, yes thank you, that definitely gives me a good starting point for getting it to honor the URL path for the "always open in" functionality. I was also going to try and modify the GUI so that we could add or modify URLs in the "always open in" list from the "edit site list" screen.

**Comment by barthel** [comment]
Any progress here?

**Comment by TriMoon** [comment]
Because #719 has been closed by @dannycolin with referencing this issue, lets see if the developers pay any attention to these split issues instead of just redirecting to new ones...
Still waiting on progress...

**Comment by crozone** [comment]
This would be especially useful since `maps.google.com` has change to `google.com/maps`, and it looks like Google is moving more and more services back into their root domain.

This means if you want to easily isolate Google Maps from Google Search (for example, if you want to be signed into Google Maps but not Google Search), it's currently impossible for the containers to distinguish the two.

**Comment by endolith** [comment]
Is this possible in any way?  Manually editing a file?  Anything?

**Comment by dsfrederic** [comment]
@endolith have a look at https://github.com/kintesh/containerise

**Comment by crozone** [comment]
> @endolith have a look at https://github.com/kintesh/containerise

Does this extension use the same underlying container system as Firefox's native containers? It just allows more flexible activation of the containers?

**Comment by amavlyanov** [comment]
> @endolith have a look at https://github.com/kintesh/containerise

looks like this extension was slightly abandoned: threre were no releases since Feb of 2021 and no comits since March of 2023.

it would be great to have this functionality "from the box" in the mozilla's own extension.

**Comment by nisc** [comment]
I use FoxyTab for this.

![Image](https://github.com/user-attachments/assets/56f8d402-84cb-4ddb-9f9e-ee9484aeb6c5)
